### DISCUSSION TITLE
录播的mp4文件放在视频编辑软件中会产生往回跳帧情况

### DISCUSSION THREAD

**Discussion by Ponyonyope** [HMBSbige/BilibiliLiveRecordDownLoader#162] [Show and tell]
工作环境：win10 64位家庭版，n卡 studio驱动
视频编辑软件：pr 2021，达芬奇16、17
首先录播的mp4文件直接在potplayer上观看没有任何问题。
但如果放在达芬奇或者pr里，无论是画面预览还是导出成品都会有连续的往回跳帧的情况。
具体表现如这一张gif图，原视频为六十帧，我截取了八帧出来做成gif。正常画面应该是绿色荧光棒从左向右挥到底，但在视频编辑软件中如图，八帧的画面中有两帧往前跳帧。这种情况会连续存在于视频的各处，导致观感很差无法直接进行视频编辑。
![pic](https://user-images.githubusercontent.com/63969532/138992792-668c3eed-63a6-40c3-9cd5-9f840dcab22d.gif)
*在达芬奇中经常出现无法解码录播mp4文件而无法导出的情况。
*不同直播间的录播都有这种情况。
*目前没试过其他录播机有没有同样的情况。
做过的尝试：
1、保证项目帧率与原视频帧率一致，无效。
2、帧插值方法用过帧混合，帧采样和光流法，无效。
3、把mp4文件抽取重新封装，以修复时间戳，无效。
4、确保项目为逐行扫描，无效。
5、视频编辑软件关闭硬件加速，更新重装显卡驱动，无效。
6、把不常见帧率（59.999帧）转换成常见帧率（如60帧），无效。
7、转换mp4为其他各种不同封装；借助其他软件修复时间戳，无效。
8、利用ffmpeg，x264 32-8bit编码器压制一遍，有效，在视频编辑器中预览或导出都正常，没有跳帧情况，有效。

**Comment by HMBSbige** [comment]
一般都是 B 站传输过来的源数据的问题，而且一般这种问题对于录播软件难以处理
